azure_costs_exporter/allocated_vm_collector.py

```diff
@@ -0,0 +1,90 @@
+from prometheus_client.core import GaugeMetricFamily
+
+from pandas import DataFrame
+
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.compute import ComputeManagementClient
+
+
+_BASE_COLUMNS = ['subscription', 'location', 'resource_group', 'vm_size']
+_COUNT_COLUMN = ['total']
+_ALL_COLUMNS = _BASE_COLUMNS + _COUNT_COLUMN
+
+
+def _extract_resource_group(vm_id):
+    return vm_id.split('/')[4]
+
+class AzureAllocatedVMCollector(object):
+    """
+    Class to export information about allocated Azure virtual machines 
+    in Prometheus compatible format.
+    """
+
+    def __init__(self, application_id, application_secret, tenant_id, subscription_ids, metric_name):
+        """
+        Constructor. Access is granted to what Microsoft calls a service principal / Azure Active Directory
+        Application / app registration. Read more about this topic at
+        https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal.
+        This page will guide you how to obtain an application_id, and application_secret, and the tenant_id of
+        your Azure Active Directory. Please do not forget to grant "Reader" permissions to the app for all
+        subscriptions that you want to monitor.
+        
+        :param application_id: The application ID that is created during the Azure app registration.
+        :param application_secret: The application secret that is created during the Azure app registration.
+        :param tenant_id: The ID of your Azure Active Directory instance
+        :param subscription_ids: A _sequence_ of subscription IDs that shall be monitored. The application_id
+                                 required "Reader" permissions on each subscription.
+        :param metric_name: Name of the timeseries
+        """
+        self._metric_name = metric_name
+        self._subscription_ids = subscription_ids
+        self._credentials = ServicePrincipalCredentials(client_id=application_id,
+                                                        secret=application_secret,
```

Comment:
We normally try to avoid indentation like this as it leads to very jagged code. Instead we move it to a new line with a consistent indentation.

For further details, see https://github.com/twitter/commons/blob/master/src/python/twitter/common/styleguide.md#best-practices

(this implies to this code here, but also many other places in the PR)

Reply:
done